<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Getting Started &#8212; BioVida 0.1.0 documentation</title>
    
    <link rel="stylesheet" href="_static/basic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="_static/bootswatch-3.3.6/simplex/bootstrap.min.css" type="text/css" />
    <link rel="stylesheet" href="_static/bootstrap-sphinx.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '0.1.0',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true,
        SOURCELINK_SUFFIX: '.txt'
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script>
    <script type="text/javascript" src="_static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="_static/bootstrap-3.3.6/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="_static/bootstrap-sphinx.js"></script>
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="API" href="API.html" />
    <link rel="prev" title="Automated BioMedical Data Curation" href="index.html" />
<meta charset='utf-8'>
<meta http-equiv='X-UA-Compatible' content='IE=edge,chrome=1'>
<meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1'>
<meta name="apple-mobile-web-app-capable" content="yes">

  </head>
  <body role="document">

  <div id="navbar" class="navbar navbar-default navbar-fixed-top">
    <div class="container">
      <div class="navbar-header">
        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>
        <a class="navbar-brand" href="index.html">
          BioVida</a>
        <span class="navbar-text navbar-version pull-left"><b>0.1</b></span>
      </div>

        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            
                <li><a href="https://github.com/TariqAHassan/BioVida">View on GitHub</a></li>
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="index.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul class="current">
<li class="toctree-l1 current"><a class="current reference internal" href="#">Getting Started</a></li>
<li class="toctree-l1"><a class="reference internal" href="#installation">Installation</a></li>
<li class="toctree-l1"><a class="reference internal" href="#dependencies">Dependencies</a></li>
<li class="toctree-l1"><a class="reference internal" href="#image-data">Image Data</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#cancer-imaging-archive">Cancer Imaging Archive</a></li>
<li class="toctree-l2"><a class="reference internal" href="#open-i-biomedical-image-search-engine">Open-i BioMedical Image Search Engine</a></li>
<li class="toctree-l2"><a class="reference internal" href="#automated-image-data-cleaning">Automated Image Data Cleaning</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="#genomic-data">Genomic Data</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#data-harvesting">Data Harvesting</a></li>
<li class="toctree-l2"><a class="reference internal" href="#exploring-available-databases">Exploring Available Databases</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="#diagnostic-data">Diagnostic Data</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#id1">Data Harvesting</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="API.html">API</a><ul>
<li class="toctree-l2"><a class="reference internal" href="API.html#module-biovida.images.openi_interface">Images</a><ul>
<li class="toctree-l3"><a class="reference internal" href="API.html#open-i-interface">Open-i Interface</a></li>
<li class="toctree-l3"><a class="reference internal" href="API.html#cancer-imaging-archive-interface">Cancer Imaging Archive Interface</a></li>
<li class="toctree-l3"><a class="reference internal" href="API.html#image-processing">Image Processing</a></li>
<li class="toctree-l3"><a class="reference internal" href="API.html#image-classification">Image Classification</a></li>
<li class="toctree-l3"><a class="reference internal" href="API.html#template-matching">Template Matching</a></li>
<li class="toctree-l3"><a class="reference internal" href="API.html#border-and-edge-detection">Border and Edge Detection</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="API.html#module-biovida.genomics.disgenet_interface">Genomics</a><ul>
<li class="toctree-l3"><a class="reference internal" href="API.html#disgenet-interface">DisGeNET Interface</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="API.html#module-biovida.diagnostics.disease_ont_interface">Diagnostics</a><ul>
<li class="toctree-l3"><a class="reference internal" href="API.html#disease-ontology-interface">Disease Ontology Interface</a></li>
<li class="toctree-l3"><a class="reference internal" href="API.html#disease-symptoms-interface">Disease-Symptoms Interface</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="API.html#module-biovida.unification.unify_domains">Domain Integration</a><ul>
<li class="toctree-l3"><a class="reference internal" href="API.html#unifying-across-domains">Unifying Across Domains</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="API.html#module-biovida.images.image_cache_mgmt">Cache Manipulation</a><ul>
<li class="toctree-l3"><a class="reference internal" href="API.html#image-cache-management">Image Cache Management</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="API.html#module-biovida.support_tools.utilities">Support Tools</a><ul>
<li class="toctree-l3"><a class="reference internal" href="API.html#utilities">Utilities</a></li>
<li class="toctree-l3"><a class="reference internal" href="API.html#printing-tools">Printing Tools</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Current Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">Getting Started</a></li>
<li><a class="reference internal" href="#installation">Installation</a></li>
<li><a class="reference internal" href="#dependencies">Dependencies</a></li>
<li><a class="reference internal" href="#image-data">Image Data</a><ul>
<li><a class="reference internal" href="#cancer-imaging-archive">Cancer Imaging Archive</a></li>
<li><a class="reference internal" href="#open-i-biomedical-image-search-engine">Open-i BioMedical Image Search Engine</a></li>
<li><a class="reference internal" href="#automated-image-data-cleaning">Automated Image Data Cleaning</a></li>
</ul>
</li>
<li><a class="reference internal" href="#genomic-data">Genomic Data</a><ul>
<li><a class="reference internal" href="#data-harvesting">Data Harvesting</a></li>
<li><a class="reference internal" href="#exploring-available-databases">Exploring Available Databases</a></li>
</ul>
</li>
<li><a class="reference internal" href="#diagnostic-data">Diagnostic Data</a><ul>
<li><a class="reference internal" href="#id1">Data Harvesting</a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
              
                
  <li>
    <a href="index.html" title="Previous Chapter: Automated BioMedical Data Curation"><span class="glyphicon glyphicon-chevron-left visible-sm"></span><span class="hidden-sm hidden-tablet">&laquo; Automated Bio...</span>
    </a>
  </li>
  <li>
    <a href="API.html" title="Next Chapter: API"><span class="glyphicon glyphicon-chevron-right visible-sm"></span><span class="hidden-sm hidden-tablet">API &raquo;</span>
    </a>
  </li>
              
            
            
            
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          
        </div>
    </div>
  </div>

<div class="container">
  <div class="row">
      <div class="col-md-3">
        <div id="sidebar" class="bs-sidenav" role="complementary"><ul>
<li><a class="reference internal" href="#">Getting Started</a></li>
<li><a class="reference internal" href="#installation">Installation</a></li>
<li><a class="reference internal" href="#dependencies">Dependencies</a></li>
<li><a class="reference internal" href="#image-data">Image Data</a><ul>
<li><a class="reference internal" href="#cancer-imaging-archive">Cancer Imaging Archive</a></li>
<li><a class="reference internal" href="#open-i-biomedical-image-search-engine">Open-i BioMedical Image Search Engine</a></li>
<li><a class="reference internal" href="#automated-image-data-cleaning">Automated Image Data Cleaning</a></li>
</ul>
</li>
<li><a class="reference internal" href="#genomic-data">Genomic Data</a><ul>
<li><a class="reference internal" href="#data-harvesting">Data Harvesting</a></li>
<li><a class="reference internal" href="#exploring-available-databases">Exploring Available Databases</a></li>
</ul>
</li>
<li><a class="reference internal" href="#diagnostic-data">Diagnostic Data</a><ul>
<li><a class="reference internal" href="#id1">Data Harvesting</a></li>
</ul>
</li>
</ul>

        </div>
      </div>
    <div class="col-md-9 content">
      
  <div class="section" id="getting-started">
<h1>Getting Started<a class="headerlink" href="#getting-started" title="Permalink to this headline">¶</a></h1>
<p>This library is primarily intended to automate the collection, post-processing and integration of biomedical data stored
in public online databases. It is hoped that this effort will catalyze new insights and understanding by transforming
multiple, distinct data repositories into unified datasets which are highly amenable to data analysis.</p>
<p>BioVida aims to curate a broad range of biomedical information. In areas such as diagnostics and genomics, this
involves drawing on the work of others, such as the impressive work by the Disease Ontology and DisGeNET teams.
In the case of image data however, BioVida itself performs the &#8216;heavy lifting&#8217; involved in collecting and processing
raw data from sources. This is made possible by combining traditional programmatic solutions with recent advances
in machine learning, namely convolutional neural networks.</p>
<p>The guide below provides a brief introduction to getting started with BioVida.</p>
</div>
<hr class="docutils" />
<div class="section" id="installation">
<h1>Installation<a class="headerlink" href="#installation" title="Permalink to this headline">¶</a></h1>
<p>Latest Build:</p>
<div class="code bash highlight-default"><div class="highlight"><pre><span></span>$ pip install git+git://github.com/TariqAHassan/BioVida@master
</pre></div>
</div>
</div>
<hr class="docutils" />
<div class="section" id="dependencies">
<h1>Dependencies<a class="headerlink" href="#dependencies" title="Permalink to this headline">¶</a></h1>
<p>BioVida requires: <a class="reference external" href="https://www.crummy.com/software/BeautifulSoup/">beautiful soup</a>,
<a class="reference external" href="http://www.h5py.org">h5py</a>,
<a class="reference external" href="https://keras.io">keras</a>,
<a class="reference external" href="https://github.com/lxml/lxml">lxml</a>,
<a class="reference external" href="http://www.numpy.org">numpy</a>,
<a class="reference external" href="http://pandas.pydata.org">pandas</a>,
<a class="reference external" href="https://github.com/python-pillow/Pillow">pillow</a>,
<a class="reference external" href="https://github.com/darcymason/pydicom">pydicom</a>,
<a class="reference external" href="http://docs.python-requests.org/en/master/">requests</a>,
<a class="reference external" href="http://scikit-image.org">scikit-image</a>,
<a class="reference external" href="https://www.scipy.org">scipy</a> and
<a class="reference external" href="https://github.com/tqdm/tqdm">tqdm</a>.</p>
<p>The BioVida installer will automatically install all of these packages.</p>
<p><strong>Notes</strong>:</p>
<p>1. Keras is used to power the convolutional neural networks in this project. This has the advantage of
allowing either <a class="reference external" href="https://www.tensorflow.org">TensorFlow</a> or
<a class="reference external" href="http://deeplearning.net/software/theano/">Theano</a> to be used as a computational backend.
If neither is present at install time, BioVida will automatically install TensorFlow for you.</p>
<p>2. To use <code class="docutils literal"><span class="pre">scipy</span></code> on macOS (formerly OSX) you will need <code class="docutils literal"><span class="pre">gcc</span></code>, which can be obtained with <code class="docutils literal"><span class="pre">homebrew</span></code> via.
<code class="docutils literal"><span class="pre">$</span> <span class="pre">brew</span> <span class="pre">install</span> <span class="pre">gcc</span></code>. If you do not have <code class="docutils literal"><span class="pre">homebrew</span></code> installed, it can be installed by following the instructions
provided <a class="reference external" href="https://brew.sh">here</a>.</p>
</div>
<hr class="docutils" />
<div class="section" id="image-data">
<h1>Image Data<a class="headerlink" href="#image-data" title="Permalink to this headline">¶</a></h1>
<div class="section" id="cancer-imaging-archive">
<h2>Cancer Imaging Archive<a class="headerlink" href="#cancer-imaging-archive" title="Permalink to this headline">¶</a></h2>
<div class="code python highlight-default"><div class="highlight"><pre><span></span><span class="c1"># 1. Import the interface for the Cancer Imaging Archive</span>
<span class="kn">from</span> <span class="nn">biovida.images</span> <span class="k">import</span> <span class="n">CancerImageInterface</span>

<span class="c1"># 2. Create an Instance of the Tool</span>
<span class="n">cii</span> <span class="o">=</span> <span class="n">CancerImageInterface</span><span class="p">(</span><span class="n">YOUR_API_KEY_HERE</span><span class="p">)</span>

<span class="c1"># 3. Perform a search</span>
<span class="n">cii</span><span class="o">.</span><span class="n">search</span><span class="p">(</span><span class="n">location</span><span class="o">=</span><span class="s1">&#39;extremities&#39;</span><span class="p">)</span>

<span class="c1"># 4. Pull the data</span>
<span class="n">cdf</span> <span class="o">=</span> <span class="n">cii</span><span class="o">.</span><span class="n">pull</span><span class="p">()</span>
</pre></div>
</div>
</div>
<div class="section" id="open-i-biomedical-image-search-engine">
<h2>Open-i BioMedical Image Search Engine<a class="headerlink" href="#open-i-biomedical-image-search-engine" title="Permalink to this headline">¶</a></h2>
<div class="code python highlight-default"><div class="highlight"><pre><span></span><span class="c1"># 1. Import the Interface for the NIH&#39;s Open-i API.</span>
<span class="kn">from</span> <span class="nn">biovida.images</span> <span class="k">import</span> <span class="n">OpeniInterface</span>

<span class="c1"># 2. Create an Instance of the Tool</span>
<span class="n">opi</span> <span class="o">=</span> <span class="n">OpeniInterface</span><span class="p">()</span>

<span class="c1"># 3. Perform a general search for MRIs and CTs</span>
<span class="n">opi</span><span class="o">.</span><span class="n">search</span><span class="p">(</span><span class="n">query</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span> <span class="n">image_type</span><span class="o">=</span><span class="p">[</span><span class="s1">&#39;mri&#39;</span><span class="p">,</span> <span class="s1">&#39;ct&#39;</span><span class="p">])</span>  <span class="c1"># Results Found: 134,113.</span>

<span class="c1"># 4. Pull the data</span>
<span class="n">search_df</span> <span class="o">=</span> <span class="n">opi</span><span class="o">.</span><span class="n">pull</span><span class="p">()</span>
</pre></div>
</div>
<p>The DataFrame created above, <code class="docutils literal"><span class="pre">df</span></code>, contains data from all fields
provided by the Open-i API. <sup>†</sup> Images referenced in the DataFrame will
automatically be harvested (unless specified otherwise).</p>
<p>The values accepted by the <code class="docutils literal"><span class="pre">image_type</span></code> argument above can easily be
reviewed:</p>
<div class="code python highlight-default"><div class="highlight"><pre><span></span><span class="n">opi</span><span class="o">.</span><span class="n">options</span><span class="p">(</span><span class="n">search_parameter</span><span class="o">=</span><span class="s1">&#39;image_type&#39;</span><span class="p">)</span>
</pre></div>
</div>
<p>Additionally, searches can easily be reviewed:</p>
<div class="code python highlight-default"><div class="highlight"><pre><span></span><span class="n">opi</span><span class="o">.</span><span class="n">current_search</span>
<span class="c1"># {&#39;image_type&#39;: [&#39;mri&#39;, &#39;ct&#39;, &#39;exclude_graphics&#39;], &#39;query&#39;: &#39;&#39;}</span>

<span class="n">opi</span><span class="o">.</span><span class="n">current_search_total</span>
<span class="c1"># 134113</span>
</pre></div>
</div>
<p>By default, graphics are excluded. This can be changed with <code class="docutils literal"><span class="pre">search()</span></code>&#8216;s <code class="docutils literal"><span class="pre">exclusions</span></code> parameter.</p>
<p><sup>†</sup> <em>Note:</em> by default, data harvesting is truncated after the first 100 results.</p>
</div>
<div class="section" id="automated-image-data-cleaning">
<h2>Automated Image Data Cleaning<a class="headerlink" href="#automated-image-data-cleaning" title="Permalink to this headline">¶</a></h2>
<p>Cleaning the images which have been downloaded is extremely simple.</p>
<div class="code python highlight-default"><div class="highlight"><pre><span></span><span class="c1"># 1. Import Image Processing Tools</span>
<span class="kn">from</span> <span class="nn">biovida.images</span> <span class="k">import</span> <span class="n">ImageProcessing</span>

<span class="c1"># 2. Instantiate the Tool using the OpeniInterface Instance</span>
<span class="n">ip</span> <span class="o">=</span> <span class="n">ImageProcessing</span><span class="p">(</span><span class="n">opi</span><span class="p">)</span>

<span class="c1"># 3. Clean the Image Data</span>
<span class="n">idf</span> <span class="o">=</span> <span class="n">ip</span><span class="o">.</span><span class="n">auto</span><span class="p">()</span>

<span class="c1"># 4. Save the Cleaned Images</span>
<span class="n">ip</span><span class="o">.</span><span class="n">save</span><span class="p">(</span><span class="s2">&quot;/save/directory/&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>While the <code class="docutils literal"><span class="pre">ImageProcessing()</span></code> classes allows you to
to control the image processing more precisely if you
wish (see the documentation <a class="reference external" href="https://tariqahassan.github.io/BioVida/API.html#image-processing">here</a>), this
fully automated approach should suffice in most cases.</p>
<p><strong>Notice</strong>: <strong>This functionality is considered to be experimental</strong>.</p>
</div>
</div>
<hr class="docutils" />
<div class="section" id="genomic-data">
<h1>Genomic Data<a class="headerlink" href="#genomic-data" title="Permalink to this headline">¶</a></h1>
<div class="section" id="data-harvesting">
<h2>Data Harvesting<a class="headerlink" href="#data-harvesting" title="Permalink to this headline">¶</a></h2>
<div class="code python highlight-default"><div class="highlight"><pre><span></span><span class="c1"># 1. Create an instance of the tool</span>
<span class="kn">from</span> <span class="nn">biovida.genomics</span> <span class="k">import</span> <span class="n">DisgenetInterface</span>

<span class="c1"># 2. Create an Instance of the Tool</span>
<span class="n">dna</span> <span class="o">=</span> <span class="n">DisgenetInterface</span><span class="p">()</span>

<span class="c1"># 3. Pull a Database</span>
<span class="n">df</span> <span class="o">=</span> <span class="n">dna</span><span class="o">.</span><span class="n">pull</span><span class="p">(</span><span class="s1">&#39;curated&#39;</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="exploring-available-databases">
<h2>Exploring Available Databases<a class="headerlink" href="#exploring-available-databases" title="Permalink to this headline">¶</a></h2>
<div class="code python highlight-default"><div class="highlight"><pre><span></span><span class="n">dna</span><span class="o">.</span><span class="n">options</span><span class="p">()</span>
<span class="c1"># Available Databases:</span>
<span class="c1">#   - &#39;all&#39;</span>
<span class="c1">#   - &#39;curated&#39;</span>
<span class="c1">#   - &#39;snp_disgenet&#39;</span>

<span class="n">dna</span><span class="o">.</span><span class="n">options</span><span class="p">(</span><span class="s1">&#39;curated&#39;</span><span class="p">)</span>
<span class="c1"># - Full Name:    Curated Gene-Disease Associations</span>
<span class="c1"># - Description:  The file contains gene-disease associations from UNIPROT, CTD (human subset),</span>
<span class="c1">#                 ClinVar, Orphanet, and the GWAS Catalog.</span>
</pre></div>
</div>
<p>As with the <code class="docutils literal"><span class="pre">OpeniInterface()</span></code> class above, it is easy to gain access
to the most recent <code class="docutils literal"><span class="pre">pull</span></code> and related information.</p>
<p>The database itself:</p>
<div class="code python highlight-default"><div class="highlight"><pre><span></span><span class="n">dna</span><span class="o">.</span><span class="n">current_database</span>
</pre></div>
</div>
<p>Information about the database:</p>
<div class="code python highlight-default"><div class="highlight"><pre><span></span><span class="n">dna</span><span class="o">.</span><span class="n">current_database_name</span>
<span class="c1"># &#39;curated&#39;</span>

<span class="n">dna</span><span class="o">.</span><span class="n">current_database_full_name</span>
<span class="c1"># &#39;Curated Gene-Disease Associations&#39;</span>

<span class="n">dna</span><span class="o">.</span><span class="n">current_database_description</span>
<span class="c1"># &#39;The file contains gene-disease associations from...&#39;</span>
</pre></div>
</div>
</div>
</div>
<hr class="docutils" />
<div class="section" id="diagnostic-data">
<h1>Diagnostic Data<a class="headerlink" href="#diagnostic-data" title="Permalink to this headline">¶</a></h1>
<div class="section" id="id1">
<h2>Data Harvesting<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h2>
<div class="code python highlight-default"><div class="highlight"><pre><span></span><span class="c1"># 1. Import the Interface for DiseaseOntology.org</span>
<span class="kn">from</span> <span class="nn">biovida.diagnostics</span> <span class="k">import</span> <span class="n">DiseaseOntInterface</span>

<span class="c1"># 2. Create an Instance of the Tool</span>
<span class="n">doi</span> <span class="o">=</span> <span class="n">DiseaseOntInterface</span><span class="p">()</span>

<span class="c1"># 3. Pull the Database</span>
<span class="n">ddf</span> <span class="o">=</span> <span class="n">doi</span><span class="o">.</span><span class="n">pull</span><span class="p">()</span>
</pre></div>
</div>
<p>One can gain access to the database, by following
the approach shown above (with <code class="docutils literal"><span class="pre">ddf</span></code>) or as follows:</p>
<div class="code python highlight-default"><div class="highlight"><pre><span></span><span class="n">doi</span><span class="o">.</span><span class="n">disease_db</span>
</pre></div>
</div>
<p>It is also possible to inspect the date on which
the database was created by <em>DiseaseOntology.org:</em></p>
<div class="code python highlight-default"><div class="highlight"><pre><span></span><span class="n">doi</span><span class="o">.</span><span class="n">db_date</span>
<span class="c1"># datetime.datetime(2017, 1, 13, 0, 0)</span>
</pre></div>
</div>
<p>Following a similar approach, information on the
symptoms associated with numerous diseases can easily be obtained.</p>
<div class="code python highlight-default"><div class="highlight"><pre><span></span><span class="c1"># 1. Import the Interface for Disease-Symptoms Information</span>
<span class="kn">from</span> <span class="nn">biovida.diagnostics</span> <span class="k">import</span> <span class="n">DiseaseSymptomsInterface</span>

<span class="c1"># 2. Create an Instance of the Tool</span>
<span class="n">dsi</span> <span class="o">=</span> <span class="n">DiseaseSymptomsInterface</span><span class="p">()</span>

<span class="c1"># 3. Pull the Database</span>
<span class="n">dsdf</span> <span class="o">=</span> <span class="n">doi</span><span class="o">.</span><span class="n">pull</span><span class="p">()</span>
</pre></div>
</div>
</div>
</div>


    </div>
      
  </div>
</div>
<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
    </p>
    <p>
        &copy; Copyright 2017, Tariq A. Hassan.<br/>
      Created using <a href="http://sphinx-doc.org/">Sphinx</a> 1.5.3.<br/>
    </p>
  </div>
</footer>
  </body>
</html>